# Get started by importing numpy
## sub heading


In [17]:
import math
import numpy as np
rng = np.random.default_rng()
array = np.array(rng.integers(low=1, high=10, size=(1000000,10)))


In [19]:
%%timeit

for i in range(1000000):
    for j in range(10):
        math.log(array[i, j])

1.21 s ± 47.5 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


In [20]:
%%timeit

np.log(array)

42.7 ms ± 231 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


## Explaination 

The python code took 1.21 seconds per loop and the numpy code took 42.7 milliseconds per loop. My numpy code ran around 28 times faster than the python code.
